src/modules/Builder/pages/BuilderApplet/BuilderApplet.tsx

```diff
@@ -160,10 +156,7 @@ export const BuilderApplet = () => {
           <>
             {isLoading && <Spinner />}
             <LinkedTabs hiddenHeader={hiddenHeader} tabs={getAppletTabs(tabErrors)} isBuilder />
-            <SaveAndPublish
-              hasPrompt={isDirty || isFromLibrary}
-              setIsFromLibrary={setIsFromLibrary}
-            />
+            {saveAndPublish}
```

Comment:
why not add check without an extra variable?
```suggestion
            {!isPerformanceTask ?? <SaveAndPublish />}
```